google/cloud/internal/grpc_opentelemetry_test.cc

```diff
@@ -104,6 +107,44 @@ TEST(OpenTelemetry, InjectTraceContextGrpc) {
   testing_util::ValidatePropagator(context);
 }
 
+struct IsInitialMetadataReadyTestCase {
+  StatusCode code;
+  bool expected;
+};
+using IsInitialMetadataReadyTest =
+    TestWithParam<IsInitialMetadataReadyTestCase>;
+
+TEST_P(IsInitialMetadataReadyTest, Validate) {
+  IsInitialMetadataReadyTestCase const& test_case = GetParam();
+
+  ASSERT_EQ(IsInitialMetadataReady(test_case.code), test_case.expected);
+}
+
+INSTANTIATE_TEST_SUITE_P(
```

Comment:
comment: IMO the set up for parameterized tests is too complicated to add value. We can accomplish the same thing with one test that has a for loop and scoped logging. :shrug: